config/spotbugs-exclude.xml

```diff
@@ -124,6 +124,12 @@
     <Field name="$jacocoData"/>
     <Bug pattern="SE_TRANSIENT_FIELD_NOT_RESTORED"/>
   </Match>
+  <Match>
+    <!-- Must scan entire list to find the best item. -->
+    <Class name="com.puppycrawl.tools.checkstyle.checks.imports.ImportOrderCheck"/>
+    <Method name="getGroupNumber"/>
```

Comment:
https://github.com/checkstyle/checkstyle/blob/40fa53c04064e8d9b46fd097eb3b639cfe7d4f63/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java#L1099
Not seeing how we can add a break here since we are looking for the best option. We can't guarantee one regexp will be better than another without executing it.

Reply:
This can be rewritten with Streams API:
```java
        return IntStream.range(0, patterns.length)
                .mapToObj(index -> Map.entry(index, patterns[index].matcher(name)))
                .filter(entry -> entry.getValue().matches())
                .min((left, right) -> {
                    int result = left.getValue().start() - right.getValue().start();
                    if (result == 0) {
                        result = right.getValue().end() - left.getValue().end();
                    }
                    return result;
                })
                .map(Map.Entry::getKey)
                .orElse(patterns.length);
```
But it will be less readable and less efficient. In my opinion, the existing implementation is better.
@strkkk could you suggest a more readable implementation of this method?

Reply:
It can be simplified a bit with Comparator.comparingInt().thenComparing()

But I would say that it is better to leave existing implementation